### PR TITLE
Adding example.

### PR DESCRIPTION
**Pull request by ThePedroo**

### CODE REVIEW

File: docs/source/discord/resources/channel/methods/add_thread_member.rst [hunk 1]
```diff
@@ -11,3 +11,12 @@ Add Thread Member
...
 
 .. doxygenfunction:: discord_add_thread_member
+
+
+Example
+-------
+
+.. code:: c
+   
+   // The channel that you executed the channel must be a thread.
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/add_thread_member.rst:21]:
```suggestion
   // channel_id must be from a thread channel
```

File: docs/source/discord/resources/channel/methods/create_channel_invite.rst [hunk 2]
```diff
@@ -12,3 +12,20 @@ Create Channel Invite
...
+   struct discord_invite ret;
+   char text[124];
+   
+   discord_create_channel_invite(client, msg->channel_id, NULL, &ret);
+   
+   snprintf(text, sizeof(text), "Created invite: %s", ret.code);
+   
+   struct discord_create_message_params params = { .content = text };
+
+   discord_async_next(client, NULL);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/create_channel_invite.rst:30]:
```suggestion
```

File: docs/source/discord/resources/channel/methods/create_message.rst [hunk 3]
```diff
@@ -13,3 +13,13 @@ Create Message
...
 .. doxygenstruct:: discord_create_message_params
+
+Example
+-------
+
+.. code:: c
+   
+   struct discord_create_message_params params = { .content = "Pong!" };
+      
+   discord_async_next(client, NULL);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/create_message.rst:24]:
```suggestion
```

File: docs/source/discord/resources/channel/methods/crosspost_message.rst [hunk 4]
```diff
@@ -12,3 +12,13 @@ Crosspost Message
...
+
+Example
+-------
+
+.. code:: c
+
+   struct discord_message ret;
+   
+   discord_crosspost_message(client, msg->channel_id, msg->id, &ret);
+
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/crosspost_message.rst:24]:
```suggestion
```

File: docs/source/discord/resources/channel/methods/delete_all_reactions.rst [hunk 5]
```diff
@@ -12,3 +12,13 @@ Delete All Reactions
...
 .. doxygenfunction:: discord_delete_all_reactions
+
+Example
+-------
+
+.. code:: c
+     
+   u64_snowflake_t message_id = strtoull(msg->content, NULL, 10);
+       
+   struct discord_channel ret;
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/delete_all_reactions.rst:23]:
```suggestion
```

File: docs/source/discord/resources/channel/methods/delete_all_reactions_for_emoji.rst [hunk 6]
```diff
@@ -12,3 +12,14 @@ Delete All Reactions For Emoji
...
+
+Example
+-------
+
+.. code:: c
+   
+   discord_create_reaction(client, msg->channel_id, msg->id, 0, "😄");
+
+   sleep(10000);
+
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/delete_all_reactions_for_emoji.rst:24]:
```suggestion
```

File: docs/source/discord/resources/channel/methods/delete_channel_permission.rst [hunk 7]
```diff
@@ -11,3 +11,12 @@ Delete Channel Permission
 =========================
 
 .. doxygenfunction:: discord_delete_channel_permission
+
+Example
+-------
+
+.. code:: c
+
+   u64_snowflake_t id = 1231231232131ULL;
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/delete_channel_permission.rst:20]:
```suggestion
   u64_snowflake_t id = 1231231232131ULL; // ficticious channel ID
```

File: docs/source/discord/resources/channel/methods/edit_channel_permissions.rst [hunk 8]
```diff
@@ -12,3 +12,12 @@ Edit Channel Permissions
 
 .. doxygenfunction:: discord_edit_channel_permissions
 .. doxygenstruct:: discord_edit_channel_permissions_params
+
+Example
+-------
+
+.. code:: c
+
+   u64_snowflake_t id = 1231231232131ULL;
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/edit_channel_permissions.rst:21]:
```suggestion
   u64_snowflake_t id = 1231231232131ULL; // ficticious channel ID
```

File: docs/source/discord/resources/channel/methods/edit_message.rst [hunk 9]
```diff
@@ -13,3 +13,18 @@ Edit Message
...
+
+   struct discord_message ret;
+   
+   struct discord_create_message_params params = { .content = "Hello there?" };
+   
+   discord_create_message(client, msg->channel_id, &params, &ret);
+   
+   struct discord_create_message_params paramss = { .content = "Hi!\nOh! What a fast response!" };
+   
+   discord_edit_message(client, msg->channel_id, ret.id, &paramss, NULL);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/edit_message.rst:30]:
```suggestion
   discord_create_message(client, 
                          msg->channel_id,
                          &(struct discord_create_message_params){ 
                            .content = "Hello there?" 
                          },
                          &ret);
   
   discord_edit_message(client,
                        msg->channel_id,
                        ret.id,
                        &(struct discord_edit_message_params){
                          .content = "Such a fast response!"
                        },
                        NULL);
```

File: docs/source/discord/resources/channel/methods/follow_news_channel.rst [hunk 10]
```diff
@@ -12,3 +12,12 @@ Follow News Channel
 
 .. doxygenfunction:: discord_follow_news_channel
 .. doxygenstruct:: discord_follow_news_channel_params
+
+Example
+-------
+
+.. code:: c
+
+   u64_snowflake_t id = 1231231232131ULL;
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/follow_news_channel.rst:21]:
```suggestion
   u64_snowflake_t id = 1231231232131ULL; // ficticious channel ID
```

File: docs/source/discord/resources/channel/methods/get_channel.rst [hunk 11]
```diff
@@ -12,3 +12,22 @@ Get Channel
...
 
 .. doxygenfunction:: discord_get_channel
+
+Example
+-------
+
+.. code:: c
+   
+   struct discord_channel ret;
+   char text[128]
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/get_channel.rst:22]:
```suggestion
```

File: docs/source/discord/resources/channel/methods/get_channel.rst [hunk 12]
```diff
@@ -12,3 +12,22 @@ Get Channel
...
+
+   discord_get_channel(client, msg->channel_id, &ret);
+
+   snprint(text, sizeof(text), "The name of this channel is: %s", ret.name)
+
+   struct discord_create_message_params params = { .content = text };
+      
+   discord_async_next(client, NULL);
+   discord_create_message(client, msg->channel_id, &params, NULL);
+
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/get_channel.rst:32]:
```suggestion
```

File: docs/source/discord/resources/channel/methods/get_channel.rst [hunk 13]
```diff
@@ -12,3 +12,22 @@ Get Channel
...
+-------
+
+.. code:: c
+   
+   struct discord_channel ret;
+   char text[128]
+
+   discord_get_channel(client, msg->channel_id, &ret);
+
+   snprint(text, sizeof(text), "The name of this channel is: %s", ret.name)
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/get_channel.rst:26]:
```suggestion
   printf("The name of this channel is: %s", ret.name)
```

File: docs/source/discord/resources/channel/methods/unpin_message.rst [hunk 14]
```diff
@@ -11,3 +11,12 @@ Unpin Message
 =============
 
 .. doxygenfunction:: discord_unpin_message
+
+Example
+-------
+
+.. code:: c
+   
+   u64_snowflake_t id = 1231231232131ULL;
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/unpin_message.rst:20]:
```suggestion
   u64_snowflake_t id = 1231231232131ULL; // ficticious channel ID
```

File: docs/source/discord/resources/channel/methods/remove_thread_member.rst [hunk 15]
```diff
@@ -11,3 +11,10 @@ Remove Thread Member
 ====================
 
 .. doxygenfunction:: discord_remove_thread_member
+
+Example
+-------
+
+.. code:: c
+   
+   discord_remove_thread_member(client, msg->channel_id, msg->author.id);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/remove_thread_member.rst:20]:
```suggestion
   discord_remove_thread_member(client, msg->channel_id, msg->author->id);
```

File: docs/source/discord/resources/channel/methods/edit_message.rst (hunk 9 above)
Comment (ThePedroo) [reply docs/source/discord/resources/channel/methods/edit_message.rst:30]: Changing some things.

File: docs/source/discord/resources/channel/methods/leave_thread.rst [hunk 16]
```diff
@@ -11,3 +11,12 @@ Leave Thread
 ============
 
 .. doxygenfunction:: discord_leave_thread
+
+Example
+-------
+
+.. code:: c
+   
+   u64_snowflake_t id = 1231231232131ULL;
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/leave_thread.rst:20]:
```suggestion
   u64_snowflake_t id = 1231231232131ULL; // ficticious channel ID
```

File: docs/source/discord/resources/channel/methods/join_thread.rst [hunk 17]
```diff
@@ -11,3 +11,12 @@ Join Thread
 ===========
 
 .. doxygenfunction:: discord_join_thread
+
+Example
+-------
+
+.. code:: c
+
+   u64_snowflake_t id = 1231231232131ULL;
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/join_thread.rst:20]:
```suggestion
   u64_snowflake_t id = 1231231232131ULL; // ficticious channel ID
```

File: docs/source/discord/resources/channel/methods/get_reactions.rst [hunk 18]
```diff
@@ -13,3 +13,20 @@ Get Reactions
...
+-------
+
+.. code:: c
+
+   struct discord_user **ret;
+
+   discord_create_reaction(client, msg->channel_id, msg->id, 0, "😄");
+   
+   sleep(5000);
+
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/get_reactions.rst:27]:
```suggestion
```

File: docs/source/discord/resources/channel/methods/get_channel_message.rst [hunk 19]
```diff
@@ -12,3 +12,22 @@ Get Channel Message
...
+   struct discord_create_message_params params = { .content = "Hey!" };
+
+   discord_create_message(client, msg->channel_id, &params, &rett);
+   
+   discord_get_channel_message(client, msg->channel_id, rett.id, &ret);
+   
+   printf("I said: %s", rett.content);
+   
+   discord_message_cleanup(ret);
+   discord_message_cleanup(rett);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/get_channel_message.rst:33]:
```suggestion
   struct discord_message ret1;
   struct discord_message ret2;

   struct discord_create_message_params params = { .content = "Hey!" };

   discord_create_message(client, msg->channel_id, &params, &ret1);
   
   discord_get_channel_message(client, msg->channel_id, ret1.id, &ret2);
   
   printf("I said: %s", ret2.content);
   
   discord_message_cleanup(&ret1);
   discord_message_cleanup(&ret1);
```

Comment (ThePedroo) [reply docs/source/discord/resources/channel/methods/get_channel_message.rst:33]: Rename variable's name

File: docs/source/discord/resources/channel/methods/add_thread_member.rst [hunk 20]
```diff
@@ -11,3 +11,12 @@ Add Thread Member
...
 .. doxygenfunction:: discord_add_thread_member
+
+
+Example
+-------
+
+.. code:: c
+   
+   // channel_id must be from a thread channel
+   discord_add_thread_member(client, msg->channel_id, msg->author.id);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/add_thread_member.rst:22]:
```suggestion
   discord_add_thread_member(client, msg->channel_id, msg->author->id); 
```

File: docs/source/discord/resources/channel/methods/create_channel_invite.rst [hunk 21]
```diff
@@ -12,3 +12,18 @@ Create Channel Invite
...
+   
+   struct discord_invite ret;
+   char text[124];
+   
+   discord_create_channel_invite(client, msg->channel_id, NULL, &ret);
+   
+   snprintf(text, sizeof(text), "Created invite: %s", ret.code);
+   
+   struct discord_create_message_params params = { .content = text };
+   discord_create_message(client, msg -> channel_id, &params, NULL);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/create_channel_invite.rst:29]:
```suggestion
   discord_create_message(client, msg->channel_id, &params, NULL);
   
   discord_invite_cleanup(&ret);
```

File: docs/source/discord/resources/channel/methods/create_message.rst [hunk 22]
```diff
@@ -13,3 +13,12 @@ Create Message
...
 .. doxygenstruct:: discord_create_message_params
+
+Example
+-------
+
+.. code:: c
+   
+   struct discord_create_message_params params = { .content = "Pong!" };
+      
+   discord_create_message(client, msg -> channel_id, &params, NULL);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/create_message.rst:24]:
```suggestion
   struct discord_create_message_params params = { .content = "Pong!" };
   discord_create_message(client, msg->channel_id, &params, NULL);
```

File: docs/source/discord/resources/channel/methods/crosspost_message.rst [hunk 23]
```diff
@@ -12,3 +12,12 @@ Crosspost Message
...
 .. doxygenfunction:: discord_crosspost_message
+
+Example
+-------
+
+.. code:: c
+
+   struct discord_message ret;
+   
+   discord_crosspost_message(client, msg->channel_id, msg->id, &ret);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/crosspost_message.rst:23]:
```suggestion
   discord_crosspost_message(client, msg->channel_id, msg->id, &ret);
   
   discord_message_cleanup(&ret);
```

File: docs/source/discord/resources/channel/methods/delete_all_reactions.rst [hunk 24]
```diff
@@ -12,3 +12,12 @@ Delete All Reactions
 ====================
 
 .. doxygenfunction:: discord_delete_all_reactions
+
+Example
+-------
+
+.. code:: c
+     
+   u64_snowflake_t message_id = strtoull(msg->content, NULL, 10);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/delete_all_reactions.rst:21]:
```suggestion
   u64_snowflake_t message_id = strtoull(msg->content, NULL, 10); // assuming msg->content holds a message ID
```

File: docs/source/discord/resources/channel/methods/delete_channel.rst [hunk 25]
```diff
@@ -12,3 +12,13 @@ Delete Channel
...
+
+Example
+-------
+
+.. code:: c
+   
+   struct discord_channel ret;
+   discord_delete_channel(client, msg->channel_id, &ret);
+       
+   printf("Deleted channel %s", ret.name);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/delete_channel.rst:24]:
```suggestion
   printf("Deleted channel %s", ret.name);
   
   discord_channel_cleanup(&ret);
```

File: docs/source/discord/resources/channel/methods/edit_message.rst [hunk 26]
```diff
@@ -13,3 +13,25 @@ Edit Message
...
+                          },
+                          &ret);
+   
+   discord_edit_message(client,
+                        msg->channel_id,
+                        ret.id,
+                        &(struct discord_edit_message_params){
+                          .content = "Such a fast response!"
+                        },
+                        NULL);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/edit_message.rst:37]:
```suggestion
   discord_edit_message(client,
                        msg->channel_id,
                        ret.id,
                        &(struct discord_edit_message_params){
                          .content = "Such a fast response!"
                        },
                        NULL);
  
   discord_message_cleanup(&ret); 
```

File: docs/source/discord/resources/channel/methods/get_channel_messages.rst [hunk 27]
```diff
@@ -13,3 +13,16 @@ Get Channel Messages
...
+
+.. code:: c
+   
+   struct discord_message **ret;
+   
+   discord_get_channel_messages(client, msg->channel_id, NULL, &ret);
+   
+   if (ret)
+     for (int i = 0; ret[i]; ++i)
+       printf("%s\n", ret[i]->content);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/get_channel_messages.rst:28]:
```suggestion
   discord_get_channel_messages(client, msg->channel_id, NULL, &ret);
   
   if (ret)
     for (int i = 0; ret[i]; ++i)
       printf("%s\n", ret[i]->content);
  
   discord_message_list_free(ret);
```

File: docs/source/discord/resources/channel/methods/get_pinned_messages.rst [hunk 28]
```diff
@@ -11,3 +11,16 @@ Get Pinned Messages
...
+
+.. code:: c
+
+   struct discord_message **ret;
+   
+   discord_get_pinned_messages(client, msg->channel_id, &ret);
+   
+   if (ret)
+     for (int i = 0; ret[i]; ++i)
+       printf("%s\n", ret[i]->content);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/get_pinned_messages.rst:26]:
```suggestion
   discord_get_pinned_messages(client, msg->channel_id, &ret);
   
   if (ret)
     for (int i = 0; ret[i]; ++i)
       printf("%s\n", ret[i]->content);
       
   discord_message_list_free(ret);
```

File: docs/source/discord/resources/channel/methods/get_reactions.rst [hunk 29]
```diff
@@ -13,3 +13,18 @@ Get Reactions
...
+   struct discord_user **ret;
+
+   discord_create_reaction(client, msg->channel_id, msg->id, 0, "😄");
+   
+   discord_get_reactions(client, msg->channel_id, msg->id, 0, "😄", NULL, &ret);
+   
+   if (ret)
+     for (int i = 0; ret[i]; ++i)
+       printf("%s\n", ret[i]->username);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/get_reactions.rst:30]:
```suggestion
   struct discord_user **ret;

   discord_create_reaction(client, msg->channel_id, msg->id, 0, "😄");
   
   discord_get_reactions(client, msg->channel_id, msg->id, 0, "😄", NULL, &ret);
   
   if (ret)
     for (int i = 0; ret[i]; ++i)
       printf("%s\n", ret[i]->username);
       
   discord_user_list_free(ret);
```

File: docs/source/discord/resources/channel/methods/pin_message.rst [hunk 30]
```diff
@@ -11,3 +11,16 @@ Pin Message
...
+
+.. code:: c
+
+   struct discord_message ret;
+
+   struct discord_create_message_params params = { .content = "I'm important, I need to get pinned!" };
+
+   discord_create_message(client, msg->channel_id, &params, &ret);
+   
+   discord_pin_message(client, msg->channel_id, ret.id);
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/pin_message.rst:26]:
```suggestion
   discord_pin_message(client, msg->channel_id, ret.id);
   
   discord_message_cleanup(&ret);
```

File: docs/source/discord/resources/channel/methods/delete_user_reaction.rst [hunk 31]
```diff
@@ -12,3 +12,12 @@ Delete User Reaction
...
 .. doxygenfunction:: discord_delete_user_reaction
+
+Example
+-------
+
+.. code:: c
+
+   discord_create_reaction(client, msg->channel_id, msg->id, 0, "😄");
+   
+   discord_delete_user_reaction(client, msg->channel_id, msg->id, msg->id, 0, "😄");
```
Comment (lcsmuller) [inline docs/source/discord/resources/channel/methods/delete_user_reaction.rst:23]: Repeats msg->id for user_id

**Review by lcsmuller — CHANGES_REQUESTED** [review]
Great job, just a tiny change that needs to be addressed first

File: docs/source/discord/resources/channel/methods/delete_user_reaction.rst (hunk 31 above)
Comment (ThePedroo) [reply docs/source/discord/resources/channel/methods/delete_user_reaction.rst]: Done

**Review by lcsmuller — APPROVED** [review]

**Pull request merged**